Increase allowed sympy version to 1.13.3 as done in e.g. https://github.com/pytorch/pytorch/pull/148575 for 2.7+

This allows using our existing sympy 1.13.3 easyconfigs.

Author: Alexander Grund (TU Dresden)

diff --git a/setup.py b/setup.py
--- a/setup.py
+++ b/setup.py
@@ -1099,7 +1099,7 @@ def main():
         "filelock",
         "typing-extensions>=4.10.0",
         'setuptools ; python_version >= "3.12"',
-        'sympy==1.13.1 ; python_version >= "3.9"',
+        "sympy>=1.13.3",
         "networkx",
         "jinja2",
         "fsspec",
